### ISSUE TITLE
[BUG] Sub events are not showing for super event "University Housing Sign-Up 2021-2022"

### ISSUE THREAD

**Issue by dobromirdobrev** [rokwire/illinois-app#161]
**Describe the bug**
For category 'Other' the sub events for super events "University Housing Sign-Up 2021-2022" are not showing and  there is a blank space in event card UI

**To Reproduce**
Steps to reproduce the behaviour:
1. Start "Illinois" app
2. Click on 'Events'
3. Select 'Other' category
4. There is a blank space below "University Housing Sign-Up 2021-2022" event (sub events are not showing). 

**Expected behavior**
Sub events are visible

cc @sandeep-ps  @mdryankov

**Comment by dobromirdobrev** [comment]
Hi @sandeep-ps ,

The problem here was that the logic for the new event filters [#83 ] was not applied for retrieving sub events of a super event. Now this is done and the sub events are shown.

However, now there is another problem with the data that comes from the backend.

The client hits the following endpoint for loading events with category "Other" for "Next 7 days":
GET: `https://api.converge-engine.com/rokwire/v1?startDate.lte=2020-11-12T07:30:30&endDate.gte=2020-11-06T07:30:30&targetAudience=students&category=Other`

The json returned for the super event "University Housing Sign-Up 2021-2022" is:
```
...
{
  "allDay": true,
  "category": "Other",
  "contacts": [
    {
      "email": "housing@illinois.edu",
      "organization": "University Housing",
      "phone": "2173337111"
    }
  ],
  "cost": "",
  "createdBy": "clhamil@illinois.edu",
  "endDate": "Thu, 22 Apr 2021 04:59:00 GMT",
  "id": "5f7dffe87a5461000759c848",
  "imageURL": "https://api.rokwire.illinois.edu/events/5f7dfce25c55e182abaafb4f/images/5f7dfffd5c55e182abaafb72",
  "isSuperEvent": true,
  "isVirtual": true,
  "location": {
    "description": "Virtual"
  },
  "longDescription": "Our university knows how to give you the classic college experience. When you live on campus you\u2019re part of a community built to support your success, in a prime location with close access to classes and buses, and the ease of a meal plan. For more information, including Housing Sign-Up dates and the Compare Your Options tool, click the link below. \r\n\r\n",
  "name": "",
  "sponsor": "",
  "startDate": "Mon, 19 Oct 2020 05:00:00 GMT",
  "subEvents": [
    {
      "id": "5f7e0054d357a4000a754f99",
      "isFeatured": true,
      "name": "University Housing Sign-Up Same Room Phase",
      "track": "Housing"
    },
    {
      "id": "5f807363d357a4000bf40205",
      "isFeatured": true,
      "name": "University Housing Sign-Up Any Room/Any Hall Phase",
      "track": "Housing"
    },
    {
      "id": "5f80793c1b0b03000a25e91c",
      "isFeatured": true,
      "name": "University Housing Sign-Up All Illini Phase",
      "track": "Housing"
    }
  ],
  "subcategory": "",
  "tags": [
    
  ],
  "targetAudience": [
    
  ],
  "title": "University Housing Sign-Up 2021-2022",
  "titleURL": "",
  "rokwire_score": 0.66
}
...
```

After that, the client hits the following endpoint in order to retrieve the sub events of the super event:
GET: `https://api.converge-engine.com/rokwire/v1?superEventId=5f7dffe87a5461000759c848&startDate.lte=2020-11-12T07:32:43&endDate.gte=2020-11-06T07:32:43`

The json response from server is:
```
[
  {
    "allDay": false,
    "category": "Other",
    "contacts": [
      {
        "email": "housing@illinois.edu",
        "organization": "University Housing",
        "phone": "+12173337111"
      }
    ],
    "cost": "",
    "createdBy": "bing@illinois.edu",
    "endDate": "Thu, 16 Apr 2020 14:59:00 GMT",
    "id": "5f80793c1b0b03000a25e91c",
    "imageURL": "https://api.rokwire.illinois.edu/events/5f7f45fd5c55e182abaafbbb/images/5f91d27cdf5b8c5b4eb8b253",
    "isSuperEvent": false,
    "isVirtual": true,
    "location": {
      "description": "University of Illinois",
      "latitude": 40.1019523,
      "longitude": -88.2271615
    },
    "longDescription": " Housing Sign-Up 2021-2022 is open to all Illini November 16-April 15. Are you an off-campus student who wishes to return to the ease of the residence halls? Sign up to choose any available space in the undergraduate or graduate residence halls. Click the link below to get started! ",
    "name": "",
    "sponsor": "University Housing ",
    "startDate": "Mon, 16 Nov 2020 19:00:00 GMT",
    "subEvents": [
      
    ],
    "subcategory": "",
    "tags": [
      
    ],
    "targetAudience": [
      
    ],
    "title": "University Housing Sign-Up All Illini Phase ",
    "titleURL": "https://housing.illinois.edu/signup",
    "rokwire_score": 1.4
  },
  {
    "allDay": false,
    "category": "Other",
    "contacts": [
      {
        "email": "housing@illinois.edu",
        "organization": "University Housing",
        "phone": "2173337111"
      }
    ],
    "cost": "",
    "createdBy": "clhamil@illinois.edu",
    "endDate": "Mon, 26 Oct 2020 04:59:00 GMT",
    "id": "5f7e0054d357a4000a754f99",
    "imageURL": "https://api.rokwire.illinois.edu/events/5f7dfc455c55e182abaafb48/images/5f8070c45c55e182abaafc05",
    "isSuperEvent": false,
    "isVirtual": true,
    "location": {
      "description": "University of Illinois",
      "latitude": 40.1019523,
      "longitude": -88.2271615
    },
    "longDescription": "Reserved for current residents, Same Room Phase of Housing Sign-Up 2021-2022 is October 19\u201325.\r\nIf you love where you live and want to stay there, sign up to keep your current room. Miss the October 3 deadline and your room will be up for grabs for someone else. Click the link below to get started. \r\n\r\n",
    "name": "",
    "sponsor": "University Housing ",
    "startDate": "Mon, 19 Oct 2020 18:00:00 GMT",
    "subEvents": [
      
    ],
    "subcategory": "",
    "tags": [
      
    ],
    "targetAudience": [
      
    ],
    "title": "University Housing Sign-Up Same Room Phase ",
    "titleURL": "https://housing.illinois.edu/apply-now/housing-sign-up",
    "rokwire_score": 1.4
  },
  {
    "allDay": false,
    "category": "Other",
    "contacts": [
      {
        "email": "housing@illinois.edu",
        "organization": "University Housing ",
        "phone": "2173337111"
      }
    ],
    "cost": "",
    "createdBy": "clhamil@illinois.edu",
    "endDate": "Fri, 16 Apr 2021 04:59:00 GMT",
    "id": "5f807363d357a4000bf40205",
    "imageURL": "https://api.rokwire.illinois.edu/events/5f8072c55c55e182abaafc2f/images/5f8074c35c55e182abaafc40",
    "isSuperEvent": false,
    "isVirtual": true,
    "location": {
      "description": "University of Illinois",
      "latitude": 40.1019523,
      "longitude": -88.2271615
    },
    "longDescription": "Reserved for current residents, Any Room/Any Hall Phase of Housing Sign-Up 2021-2022 is November 4-April 15. Ready to try a new floor or hall? Sign up to choose any available space in the undergraduate or graduate residence halls. Click the link below to get started!\r\n",
    "name": "",
    "sponsor": "University Housing ",
    "startDate": "Wed, 04 Nov 2020 19:00:00 GMT",
    "subEvents": [
      
    ],
    "subcategory": "",
    "tags": [
      
    ],
    "targetAudience": [
      
    ],
    "title": "University Housing Sign-Up Any Room/Any Hall Phase ",
    "titleURL": "https://housing.illinois.edu/apply-now/housing-sign-up",
    "rokwire_score": 1.4
  }
]
```

Please note the `startDate` and `endDate` for these events. 
For the first event - startDate is after the endDate.
For the second event - startDate and endDate are not in the time frame of the queryParameters `startDate.lte=2020-11-12T07:32:43&endDate.gte=2020-11-06T07:32:43`

IMO, this needs to be address on the backend team.

**Comment by dobromirdobrev** [comment]
P.S. Fixed in version 2.3.18.

**Comment by sandeep-ps** [comment]
Thanks, @dobromirdobrev.

>For the first event - startDate is after the endDate.
I think this is a data issue. I will work with @kpecknold to get this resolved.

>For the second event - startDate and endDate are not in the time frame of the queryParameters startDate.lte=2020-11-12T07:32:43&endDate.gte=2020-11-06T07:32:43

Yes, I agree. We will get this fixed on the Events Building Block. @bingzhang, please see the issue above. We need subevents results also to respect the date query parameters like how the current regular events.
